airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -243,14 +244,28 @@ def _send(
                 "Receiving response", extra={"headers": response.headers, "status": response.status_code, "body": response.text}
             )
 
-        # Request/repsonse logging for declarative cdk.
+        # Request/response logging for declarative cdk
         if log_formatter is not None and response is not None and self._message_repository is not None:
             formatter = log_formatter
             self._message_repository.log_message(
                 Level.DEBUG,
                 lambda: formatter(response),  # type: ignore # log_formatter is always cast to a callable
             )
 
+        # Emit stream status RUNNING with the reason RATE_LIMITED to log that the rate limit has been reached
+        if error_resolution.response_action == ResponseAction.RATE_LIMITED:
+            # TODO: Update to handle with message repository when concurrent message repository is ready
+            reasons = [AirbyteStreamStatusReason(type=AirbyteStreamStatusReasonType.RATE_LIMITED)]
+            message = stream_status_as_airbyte_message(StreamDescriptor(name=self._name), AirbyteStreamStatus.RUNNING, reasons).json(
```

Comment:
@brianjlai can you confirm the expectation is we'll send a "RUNNING" status message with a reason "RATE_LIMITED"?

Reply:
Yes that is correct. Sample stream status rate limited message is as follows and I believe this adheres (minus the quota reset which is not required or scoped)

```
{
   "type": "TRACE",
   "trace": {
       "type": "STREAM_STATUS",
       "emitted_at": 1.714638052064E12,
       "stream_status": {
           "stream_descriptor": {
               "name": "stream-name",
               "namespace": "stream-namespace"
           },
           "status": "RUNNING",
           "reasons": [{ —-> NEW
               "type": "RATE_LIMITED",
               "rate_limited": {
                   "quota_reset": 1714638052064
               }
           }]
       }
   }
}
```